### PR TITLE
Document Persistable and Indexable

### PR DESCRIPTION
**Pull request by GalRogozinski**
# Description

Document Persistable and Indexable

Fixes #1157 

## Type of change
- Documentation Fix

### CODE REVIEW

File: src/main/java/com/iota/iri/storage/Persistable.java [hunk 1]
```diff
@@ -2,10 +2,49 @@
 
 import java.io.Serializable;
 
+/**
+ * Any object that can be serialized and stored in the DB.
```
Comment (kwek20) [inline src/main/java/com/iota/iri/storage/Persistable.java]:
```suggestion
 * Any object that can be serialized and stored in the database.
```

You write database everywhere else, should keep this constant

File: src/main/java/com/iota/iri/storage/Persistable.java [hunk 2]
```diff
@@ -2,10 +2,49 @@
...
+ *
+ * @see Indexable
+ */
 public interface Persistable extends Serializable {
+    /**
+     * Serializes the core fields of the object to bytes that can be stored in a database. Core fields are all the
+     * fields that represent the object in other nodes in the network.
```
Comment (kwek20) [inline src/main/java/com/iota/iri/storage/Persistable.java]: "the object in other nodes" is quite vague, the other nodes likely also have metadata
```suggestion
     * fields that represent the core data required to construct this object.
```

File: src/main/java/com/iota/iri/storage/Persistable.java [hunk 3]
```diff
@@ -2,10 +2,49 @@
...
+     *
+     * @return the serialized bytes that represents the core original object
+     */
     byte[] bytes();
+
+    /**
+     * Recreates the core state (populates the fields) of the object by deserializing {@code bytes}.
+     *
+     * @param bytes the serialized version the object
```
Comment (kwek20) [inline src/main/java/com/iota/iri/storage/Persistable.java]:
```suggestion
     * @param bytes the serialized version of the object
```

File: src/main/java/com/iota/iri/storage/Persistable.java [hunk 4]
```diff
@@ -2,10 +2,49 @@
...
+     */
     void read(byte[] bytes);
+
+    /**
+     * Serializes fields that usually contain data about the object that the node itself calculates
+     * and is not passed to different nodes. This data is supposed to be stored in a different column family/table then
```
Comment (kwek20) [inline src/main/java/com/iota/iri/storage/Persistable.java]:
```suggestion
     * and is not passed to different nodes. This data is stored in a different column family/table then
```
Its not supposed to right? We always do this.

File: src/main/java/com/iota/iri/storage/Persistable.java [hunk 5]
```diff
@@ -2,10 +2,49 @@
...
+
+    /**
+     * Recreates the metadata (populates the relevant fields) of the object by deserializing {@code bytes}.
+     *
+     * @param bytes the serialized bytes that represents the metadata of the object
+     */
     void readMetadata(byte[] bytes);
+
+    /**
+     * Specifies whether the object should be appended to an existing key/index or should it replace the old
```
Comment (kwek20) [inline src/main/java/com/iota/iri/storage/Persistable.java]:
```suggestion
     * Specifies whether the object should be appended to an existing key/index or should replace the old
```

**Comment by GalRogozinski** [comment]
@kwek20 
no need to capitalize

**Pull request merged**